File: spark-example/src/main/java/com/datatorrent/example/utils/ApexPartition.java
package com.datatorrent.example.utils;

import org.apache.spark.Partition;

import java.io.Serializable;

/**
 * Created by anurag on 14/12/16.
 */
public abstract class ApexPartition implements Partition,Serializable {
    @Override
    public int index() {
        return 0;
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public boolean equals(Object other) {
        return super.equals(other);
    }
}

File: spark-example/src/main/java/com/datatorrent/example/apexscala/ApexPartition.scala
package com.datatorrent.example.apexscala

import org.apache.spark.Partition

/**
  * Created by anurag on 14/12/16.
  */
class ApexPartition extends Partition {
  def index = 0

  override def hashCode: Int = super.hashCode

  override def equals(other: Any): Boolean = super.equals(other)
}
